### PR TITLE
Added exportFormat parameter to LLMChatView

### PR DESCRIPTION
**Pull request by nriedman**
# LLMChatView exportFormat parameter

## :recycle: Current situation & Problem
Currently, the LLMChatView creates an instance of the ChatView by passing .pdf as the exportFormat. The result is that there is no way to change the export format or disable exporting entirely.

## :gear: Release Notes 
Instead of passing .pdf to the ChatView, LLMChatView now has an optional parameter exportFormat that is passed to the ChatView. By default, it is .pdf (so the .init signature need not change), but can now take any of .pdf, .json, .text, or .none. If exportFormat is .none, no export button will appear in the toolbar.

## :books: Documentation
The LLMChatView/init now has an optional new signature where the user passes in a value for exportFormat.
```swift
struct LLMChatTestView: View {
      // Use the convenience property wrapper to instantiate the `LLMMockSession`
     @LLMSessionProvider(schema: LLMMockSchema()) var llm: LLMMockSession
     @State var muted = true

     var body: some View {
         LLMChatView(session: $llm, exportFormat: .none)
             .speak(llm.context, muted: muted)
             .speechToolbarButton(muted: $muted)
     }
 }
``` 
Alternatively, the exportFormat parameter may be omitted, in which case the LLMChatView will default to .pdf.

## :white_check_mark: Testing
Export functionality is rigorously tested as part of the UI testing for the Spezi ChatView.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PSchmiedmayer** [comment]
Thank you for the PR @nriedman; amazing!

Feel free to request a review from @philippzagar once you are happy with the result of the PR and would like to get a review.

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 1]
```diff
@@ -92,9 +95,13 @@ public struct LLMChatView<Session: LLMSession>: View {
     /// Creates a ``LLMChatView`` with a `Binding` of a ``LLMSession`` that provides developers with a basic chat view to interact with a Spezi LLM.
     ///
     /// - Parameters:
-    ///   - model: A `Binding` of a  ``LLMSession`` that contains the ready-to-use LLM to generate outputs based on user input.
-    public init(session: Binding<Session>) {
+    ///   - session: A `Binding` of a  ``LLMSession`` that contains the ready-to-use LLM to generate outputs based on user input.
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:98]: Thanks for fixing the docs here (so switching `model` to `session`), greatly appreciated!

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 2]
```diff
@@ -92,9 +95,13 @@ public struct LLMChatView<Session: LLMSession>: View {
...
+    ///   - exportFormat: An optional ``ChatView.ChatExportFormat`` that defines the format of the to-be-exported ``Chat``
+    ///         - Can be .pdf, .json, .text, or .none (default is .pdf)
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:100]: Sadly, the additional intended docs are not really possible within DocC

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 3]
```diff
@@ -92,9 +95,13 @@ public struct LLMChatView<Session: LLMSession>: View {
     /// Creates a ``LLMChatView`` with a `Binding` of a ``LLMSession`` that provides developers with a basic chat view to interact with a Spezi LLM.
     ///
     /// - Parameters:
-    ///   - model: A `Binding` of a  ``LLMSession`` that contains the ready-to-use LLM to generate outputs based on user input.
-    public init(session: Binding<Session>) {
+    ///   - session: A `Binding` of a  ``LLMSession`` that contains the ready-to-use LLM to generate outputs based on user input.
+    ///   - exportFormat: An optional ``ChatView.ChatExportFormat`` that defines the format of the to-be-exported ``Chat``
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:99]: Backticks

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 4]
```diff
@@ -92,9 +95,13 @@ public struct LLMChatView<Session: LLMSession>: View {
...
+    ///         - Can be .pdf, .json, .text, or .none (default is .pdf)
+    ///         -  If .none, the chatview will present no export button in the toolbar
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:101]: Reference the `ChatView` via a single backtick

**Review by philippzagar — CHANGES_REQUESTED** [review]
Thanks so much for the PR @nriedman, greatly appreciated! 🚀 
I had some minor change requests that I persisted as in-line comments. Feel free to actually build the documentation within Xcode and check out the result via `Product > Build Documentation`, that helps to resolve lots of issues I mentioned in the in-line comments 👍

File: Sources/SpeziLLM/Views/LLMChatView.swift (hunk 4 above)
Comment (nriedman) [reply Sources/SpeziLLM/Views/LLMChatView.swift:101]: It appears that replacing the '.' with a backslash does not fix the issue. Neither does replacing the expression with 'SpeziChat/ChatView/ChatExportFormat', '/ChatView/ChatExportFormat', or 'SpeziLLM/ChatView/ChatExportFormat'. Any advice here? Thanks so much for the feedback!

Comment (philippzagar) [reply Sources/SpeziLLM/Views/LLMChatView.swift:101]: What are you exactly trying to do? As stated above, one sadly isn't able to refer to types from other SPM packages via the double backtick notation. Therefore, please just use `ChatView/ChatExportFormat`, that's good enough 👍

File: Sources/SpeziLLM/SpeziLLM.docc/SpeziLLM.md [hunk 5]
```diff
@@ -160,12 +160,12 @@ struct LLMDemoView: View {
 
 The ``LLMChatView`` and ``LLMChatViewSchema`` present a basic chat views that enables users to chat with a Spezi LLM in a typical chat-like fashion. The input can be either typed out via the iOS keyboard or provided as voice input and transcribed into written text.
 The ``LLMChatViewSchema`` takes an ``LLMSchema`` instance to define which LLM in what configuration should be used for the text inference.
-The ``LLMChatView`` is passed an ``LLMSession`` that represents the LLM in execution containing state and context, and an optional ``ChatView/ChatExportFormat`` that defines the format of the to-be-exported ``Chat`` (can be any of .pdf, .text, .json, or .none, defaults to .pdf).
+The ``LLMChatView`` is passed an ``LLMSession`` that represents the LLM in execution containing state and context, and an optional ChatView/ChatExportFormat that defines the format of the to-be-exported SpeziChat/Chat (can be any of .pdf, .text, .json, or .none, defaults to .pdf).
```
Comment (philippzagar) [inline Sources/SpeziLLM/SpeziLLM.docc/SpeziLLM.md:163]: Use `SpeziChat/Chat` with one tick each

File: Sources/SpeziLLM/SpeziLLM.docc/SpeziLLM.md [hunk 6]
```diff
@@ -160,12 +160,12 @@ struct LLMDemoView: View {
...
 
 > Tip: The ``LLMChatView`` and ``LLMChatViewSchema`` build on top of the [SpeziChat package](https://swiftpackageindex.com/stanfordspezi/spezichat/documentation).
     For more details, please refer to the DocC documentation of the [`ChatView`](https://swiftpackageindex.com/stanfordspezi/spezichat/documentation/spezichat/chatview).
 
-> Tip: By defuault, the ``LLMChatView`` presents a share button in the toolbar that exports the current ``Chat``. To remove this element and disable export functionality, pass ``.none`` for the exportFormat parameter in the ``LLMChatView`` initialization.
+> Tip: By defuault, the ``LLMChatView`` presents a share button in the toolbar that exports the current SpeziChat/Chat. To remove this element and disable export functionality, pass .none for the exportFormat parameter in ``LLMChatView/init(session:exportFormat:)``.
```
Comment (philippzagar) [inline Sources/SpeziLLM/SpeziLLM.docc/SpeziLLM.md:168]: same here

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 7]
```diff
@@ -13,7 +13,7 @@ import SwiftUI
 
 /// Chat view that enables users to interact with an LLM based on an ``LLMSession``.
 ///
-/// The ``LLMChatView`` takes an ``LLMSession`` instance and an optional ``SpeziChat/ChatView/ChatExportFormat`` as parameters within the ``LLMChatView/init(session:exportFormat:)``. The ``LLMSession`` is the executable version of the LLM containing context and state as defined by the ``LLMSchema``. The ``SpeziChat/ChatView/ChatExportFormat`` defaults to .pdf, and can be any of .pdf, .text, .json, or .none.
+/// The ``LLMChatView`` takes an ``LLMSession`` instance and an optional ChatView/ChatExportFormat as parameters within the ``LLMChatView/init(session:exportFormat:)``. The ``LLMSession`` is the executable version of the LLM containing context and state as defined by the ``LLMSchema``. The ChatView/ChatExportFormat defaults to .pdf, and can be any of .pdf, .text, .json, or .none.
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:16]: same here

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 8]
```diff
@@ -28,7 +28,7 @@ import SwiftUI
 /// The next code examples demonstrate how to use the ``LLMChatView`` with ``LLMSession``s.
 ///
 /// The ``LLMChatView`` must be passed a ``LLMSession``, meaning a ready-to-use LLM, resulting in the need for the developer to manually allocate the ``LLMSession`` via the ``LLMRunner`` and ``LLMSchema`` (which includes state management).
-/// The ``LLMChatView`` may also be passed a ``ChatView/ChatExportFormat`` to determine the export format of the to-be-exported ``Chat``. This parameter may be omitted, in which case the format will be ``.pdf``. If ``.none`` is passed, no share button will be displayed in the toolbar.
+/// The ``LLMChatView`` may also be passed a ChatView/ChatExportFormat to determine the export format of the to-be-exported SpeziChat/Chat. This parameter may be omitted, in which case the format will be .pdf. If .none is passed, no share button will be displayed in the toolbar.
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:31]: same here

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 9]
```diff
@@ -58,7 +58,7 @@ public struct LLMChatView<Session: LLMSession>: View {
         llm.state.representation == .processing
     }
     
-    /// Defines the export format of the to-be-exported ``Chat``
+    /// Defines the export format of the to-be-exported SpeziChat/Chat
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:61]: same

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 10]
```diff
@@ -96,9 +96,7 @@ public struct LLMChatView<Session: LLMSession>: View {
     ///
     /// - Parameters:
     ///   - session: A `Binding` of a  ``LLMSession`` that contains the ready-to-use LLM to generate outputs based on user input.
-    ///   - exportFormat: An optional ``ChatView.ChatExportFormat`` that defines the format of the to-be-exported ``Chat``
-    ///         - Can be .pdf, .json, .text, or .none (default is .pdf)
-    ///         -  If .none, the chatview will present no export button in the toolbar
+    ///   - exportFormat: An optional ChatView/ChatExportFormat that defines the format of the to-be-exported SpeziChat/Chat (defaults to .pdf)
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:99]: same here

**Review by philippzagar — CHANGES_REQUESTED** [review]
Thanks @nriedman, just minor fixes left

File: Sources/SpeziLLM/SpeziLLM.docc/SpeziLLM.md [hunk 11]
```diff
@@ -158,13 +158,15 @@ struct LLMDemoView: View {
 
 ### LLM Chat View
 
-The ``LLMChatView`` and ``LLMChatViewSchema`` present a basic chat views that enables users to chat with a Spezi LLM in a typical chat-like fashion. The input can be either typed out via the iOS keyboard or provided as voice input and transcribed into written text.
+The ``LLMChatView`` and ``LLMChatViewSchema`` present basic chat views that enable users to chat with a Spezi LLM in a typical chat-like fashion. The input can be either typed out via the iOS keyboard or provided as voice input and transcribed into written text.
```
Comment (philippzagar) [inline Sources/SpeziLLM/SpeziLLM.docc/SpeziLLM.md:161]: Thanks for cleaning up the docs!

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 12]
```diff
@@ -92,9 +94,14 @@ public struct LLMChatView<Session: LLMSession>: View {
     /// Creates a ``LLMChatView`` with a `Binding` of a ``LLMSession`` that provides developers with a basic chat view to interact with a Spezi LLM.
     ///
     /// - Parameters:
-    ///   - model: A `Binding` of a  ``LLMSession`` that contains the ready-to-use LLM to generate outputs based on user input.
-    public init(session: Binding<Session>) {
+    ///   - session: A `Binding` of a  ``LLMSession`` that contains the ready-to-use LLM to generate outputs based on user input.
+    ///   - exportFormat: An optional `ChatView/ChatExportFormat` to enable the chat export functionality and define the format of the to-be-exported `SpeziChat/Chat`.
+    public init(
+        session: Binding<Session>,
+        exportFormat: ChatView.ChatExportFormat? = nil
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:101]: We should also include this configuration option in the init of: https://github.com/StanfordSpezi/SpeziLLM/blob/main/Sources/SpeziLLM/Views/LLMChatViewSchema.swift

File: Sources/SpeziLLM/Views/LLMChatView.swift [hunk 13]
```diff
@@ -42,7 +43,7 @@ import SwiftUI
 ///     @State var muted = true
 ///
 ///     var body: some View {
-///         LLMChatView(session: $llm)
+///         LLMChatView(session: $llm, exportFormat: .pdf)
```
Comment (philippzagar) [inline Sources/SpeziLLM/Views/LLMChatView.swift:46]: 🚀

**Review by philippzagar — CHANGES_REQUESTED** [review]
@nriedman Only one small adjustment necessary, then you should be good to go! 🚀

**Review by philippzagar — APPROVED** [review]
Thanks @PSchmiedmayer for the final changes!

**Pull request merged**